paymentsheet/src/main/java/com/stripe/android/paymentsheet/BasePaymentMethodsListFragment.kt

```diff
@@ -127,26 +128,38 @@ internal abstract class BasePaymentMethodsListFragment(
         lifecycleScope.launch {
             repeatOnLifecycle(Lifecycle.State.RESUMED) {
                 sheetViewModel.paymentOptionsState.collect { paymentOptionsState ->
-                    adapter.update(
-                        items = paymentOptionsState.items,
-                        selectedIndex = paymentOptionsState.selectedIndex,
-                    )
+                    paymentOptionsState?.items?.let { items ->
+                        adapter.update(
+                            items = items,
+                            selectedIndex = paymentOptionsState.selectedIndex,
+                        )
+                    }
                 }
             }
         }
 
-        sheetViewModel.paymentMethods.observe(viewLifecycleOwner) { paymentMethods ->
+        sheetViewModel.paymentMethods.collectInFragment { paymentMethods ->
             if (isEditing && paymentMethods.isEmpty()) {
                 isEditing = false
             }
         }
 
-        sheetViewModel.processing.observe(viewLifecycleOwner) { isProcessing ->
+        sheetViewModel.processing.collectInFragment { isProcessing ->
             adapter.isEnabled = !isProcessing
             layoutManager.canScroll = !isProcessing
         }
     }
 
+    private inline fun <T> StateFlow<T>.collectInFragment(crossinline transform: (T) -> Unit) {
+        lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                collect {
+                    transform(it)
+                }
+            }
+        }
+    }
+
```

Comment:
This can be useful on other scenarios, and also on different lifecycles. What about extracting it to a ui utils file? like the [IOSched](https://github.com/google/iosched/blob/main/mobile/src/main/java/com/google/samples/apps/iosched/util/UiUtils.kt#L60) app does. 

```kotlin
inline fun <T> Flow<T>.launchAndCollectIn(
    owner: LifecycleOwner,
    minActiveState: Lifecycle.State = Lifecycle.State.STARTED,
    crossinline action: suspend CoroutineScope.(T) -> Unit
) = owner.lifecycleScope.launch {
        owner.repeatOnLifecycle(minActiveState) {
            collect {
                action(it)
            }
        }
    }
```